tests/ind-var-bounds-and-types.ispc

```diff
@@ -0,0 +1,40 @@
+#include "../test_static.isph"
+task void f_v(uniform float RET[]) {
+    uniform int errorCount = 0;
+
+    for (unsigned int8 a = 1; a < 255; ++a) {
+        for (uniform unsigned int8 b = 1; b < 64; b *= 2) {
```

Comment:
1. No need for inner loop. The property that we'd like to keep is just outer loop iteration space / induction variable type.
2. What'about keeping all the types? [u]int[8|16|32]?
3. The check inside must be preferably the cheapest possible. `pow(double, double)` is not cheap.
4. It's ok to remove any control flow from the loop, as we can speed up the test this way. I.e. verifying just the end result should ok.

Reply:
Just in case, I don't thnik that https://github.com/ispc/ispc/pull/2574/commits/7a27f6bea740769aef105476a283a3899186bc1a addresses any of these concerns.

Reply:
Right - updating now. 7a27f6b was a fix for the computation in the previous version

Reply:
What do you think of the new test? 5ee16db55e7f23ed3f81eeb92daf836c828b6769

Reply:
If adding 64 bit types, start not from 0, but close to the end of the range. Same for 32 bits to reduce the runtime.

Please report the runtime  for the test that you observe. Both `-O2` and `-O0`.

Reply:
Updated:
* O0: 0.013 s 
* O2: 0.006 s